### PR TITLE
Adding optional cluster parameters

### PR DESCRIPTION
**Pull request by bcicen**
Adds new cluster parameters with defaults as described [here](https://github.com/coreos/etcd/blob/master/Documentation/configuration.md).

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -67,4 +67,9 @@
   # Tuning settings
   $etcd_max_result_buffer       = '1024'
   $etcd_max_retry_attempts      = '3'
+
+	# Cluster settings
+	$etcd_cluster_active_size          = false
+	$etcd_cluster_remove_delay         = '1800.0'
```
Comment (solarkennedy) [inline manifests/params.pp]: Looks like too much whitespace?

File: templates/etcd.conf.erb [hunk 2]
```diff
@@ -33,3 +33,10 @@ cert_file          = "<%= @peer_cert_File -%>"
 key_file           = "<%= @peer_key_file -%>"
 election_timeout   = <%=  @peer_election_timeout %>
 heartbeat_interval = <%=  @peer_heartbeat_interval %>
+
+[cluster]
+<% if @cluster_active_size -%>
```
Comment (solarkennedy) [inline templates/etcd.conf.erb:38]: I would like to see unit tests on these new parameters

**Comment by bcicen** [comment]
Sure, fixed whitespace and updated unit tests for sysconfig and etcd.conf.

File: templates/etcd.sysconfig.erb [hunk 3]
```diff
@@ -73,6 +73,12 @@ RAFT_CAFILE="<%= @peer_ca_file -%>"
 RAFT_CERT="<%= @peer_cert_File -%>"
 RAFT_KEY="<%= @peer_key_file -%>"
 
+# Initial cluster settings
+# After the cluster is running, only the admin endpoint is used
+<% if @cluster_active_size -%>CLUSTER_ACTIVE_SIZE="<%= @cluster_active_size %>"<% end -%>
```
Comment (solarkennedy) [inline templates/etcd.sysconfig.erb:78]: It looks like this needs to _not_ be here if cluster_active_size is false, which it is by default? Is this correct?

If so, can you add a test case to ensure this line is not in the appropriate files in the default context?

Comment (bcicen) [inline templates/etcd.sysconfig.erb:78]: Correct, it the line should be absent by default. I might be missing something, but I believe that test is covered by my updates to the default configs in fixtures by the test at https://github.com/bcicen/puppet-etcd/blob/master/spec/classes/etcd_spec.rb#L275

Comment (solarkennedy) [inline templates/etcd.sysconfig.erb:78]: Ah got it, you are right. Thank you for confirming.

**Comment by solarkennedy** [comment]
Thank you!

**Pull request merged**